lib/codecs/src/encoding/format/raw_message.rs

```diff
@@ -48,7 +48,7 @@ impl Encoder<Event> for RawMessageSerializer {
                 .get_by_meaning(message_key)
                 .or_else(|| log.get(message_key)) // backward compatibility
                 .map(|value| value.coerce_to_bytes()),
-            Event::Metric(_) => None,
+            Event::Metric(metric) => Some(metric.to_string().into()),
```

Comment:
Do we want to support metrics in `RawMessageSerializer`? This seems at odds with [the sink input type PR](https://github.com/vectordotdev/vector/pull/12561/files#diff-49ff9ead8250b219380497e0df68eea7c79e8c2e8af9b3bcffe42507f5eb7153=).

Reply:
Yeah, thinking about this more. I think there are two serializers here: `binary` and `text`. For `binary` only logs should be supported. `text` should support encoding metrics for use with the `console` sink (and others). I think OpenMetrics, as we do now, is a good option.

Reply:
This has been resolved with #12754, going to update this PR accordingly